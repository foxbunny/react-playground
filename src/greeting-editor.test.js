const greetingEditor = require('./greeting-editor')
const { render, fireEvent, cleanup } = require('react-testing-library')

afterEach(cleanup)

describe('greetingEditor', () => {
  test('basic rendering', () => {
    const el = greetingEditor({
      name: 'John',
      reset: jest.fn(),
      updateName: jest.fn()
    })
    const { asFragment } = render(el)
    expect(asFragment()).toMatchSnapshot()
  })

  test('trigger update', () => {
    const f = jest.fn()
    const el = greetingEditor({
      name: 'John',
      reset: jest.fn(),
      updateName: f
    })
    const { getByLabelText } = render(el)
    fireEvent.change(getByLabelText('Name:'), { target: { value: 'Changed' } })
    expect(f).toHaveBeenCalled()
  })

  test('trigger reset', () => {
    const f = jest.fn()
    const el = greetingEditor({
      name: 'John',
      reset: f,
      updateName: jest.fn()
    })
    const { getByText } = render(el)
    fireEvent.click(getByText('Reset'))
    expect(f).toHaveBeenCalled()
  })
})
