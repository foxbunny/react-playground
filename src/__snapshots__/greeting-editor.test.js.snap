// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`greetingEditor basic rendering 1`] = `
<DocumentFragment>
  <label>
    Name:
    <input
      value="John"
    />
  </label>
  <p>
    Hello, John!
  </p>
  <div>
    <button>
      Reset
    </button>
  </div>
</DocumentFragment>
`;
